specifyweb/frontend/js_src/lib/components/SchemaConfig/Fields.tsx

```diff
@@ -32,6 +32,7 @@ export function SchemaConfigFields({
     sortedItems,
     (item) => table.getField(item.name)!.isRelationship
   );
+  const [isHiddenFirst, setIsHiddenFirst] = React.useState(false);
```

Comment:
Your change only fixed this for fields, not for relationships.
Also, your change caused some code duplication.
Consider how this can be simplified and improved by sorting the fields instead